docs/2_1_common_api.adoc

```diff
@@ -1108,6 +1110,63 @@ To get the directional derivative of latexmath:[{Y}] with respect to latexmath:[
 Note that in order to get the directional derivative of latexmath:[{Y}] with respect to latexmath:[{U(2,1)}], the seed value `{0, 0, 1.0, 0}` has been used.
 The retrieved directional derivative `dd` is stored in a matrix of size 3x2, so `nDvUnknown` is 6.
 
+===== Adjoint derivatives
+
+If the capability attribute `providesAdjointDerivative = true`, <<fmi3GetAdjointDerivative>> computes the adjoint deriatives of an FMU.
+
+_[Adjoint derivatives are beneficial in several contexts:_
+
+* _in artifical intelligence (AI) frameworks the adjoint derivatives are called "vector gradient products" (VJPs)._
+_There adjoint derivatives are used in the backpropagation process to perform gradient-based optimization of parameters using reverse mode automatic differentiation (AD), see, e.g., <<BPRS15>>._
+
+* _in pararameter estimation (see <<BKF17>>)_
+
+_Typically, reverse mode automatic differentiatian (AD) is more efficient for these use cases than forward mode AD, as explained in the cited references._
+
+_If one would like to construct the full Jacobian matrix, one can use either <<fmi3GetDirectionalDerivative>> (to column-wise construct it) or <<fmi3GetAdjointDerivative>> (to row-wise construct it, possibly improved with coloring methods as mentioned above)._
+_However in the applications motivating the adjoint derivatives, one does not need the full Jacobian matrix latexmath:[\mathbf{J}], but vector  latexmath:[\mathbf{J}] multiplied from the left to the Jacobian, i.e. latexmath:[\mathbf{v^T J}]._
+_For computing the full Jacobian matrix the column-wise construct is generally more efficient.]_
+
+Using the same notation as for the directional derivatives for the different modes in which the FMU can be, the adjoint derivative is
```

Comment:
> for the different modes in which the FMU can be

What does that mean? States?

Reply:
> > for the different modes in which the FMU can be
> 
> What does that mean? States?

Yes. It refers to state, that we call modes like "initialization mode".
It is analogous to the text that we we have for the directional derivatives.
(Citing from https://github.com/modelica/fmi-standard/blob/master/docs/2_1_common_api.adoc:  "An FMU has different **modes** and in every **mode** an FMU might be described by different equations and different unknowns. The precise definitions are given in the mathematical descriptions of Model Exchange ([math-model-exchange]) and Co-Simulation ([math-co-simulation]). In every **mode**, the general form of the FMU equations are:"

Shall we change all of these ocurrances to "states"?

Reply:
> Shall we change all of these ocurrances to "states"?

In this PR I would use "states" instead of "modes" to avoid ambiguities.